### ISSUE TITLE
Server crashes when inserting a JSON document with the wrong type in a repeated field

### ISSUE THREAD

**Issue by jabagawee** [typesense/typesense#220]
## Steps to reproduce

```shell
export TYPESENSE_API_KEY=1234567890
mkdir /tmp/typesense-data
docker run -d -p 8108:8108 -v/tmp/typesense-data:/data typesense/typesense:0.19.0 --data-dir /data --api-key=$TYPESENSE_API_KEY
curl "http://localhost:8108/collections" -X POST -H "Content-Type: application/json" -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" -d '{
         "name": "companies",
         "fields": [                                
           {"name": "company_name", "type": "string[]" },
           {"name": "num_employees", "type": "int32" },
           {"name": "country", "type": "string", "facet": true }
         ],
         "default_sorting_field": "num_employees"
       }'
curl "http://localhost:8108/collections/companies/documents" -X POST -H "Content-Type: application/json" -H "X-TYPESENSE-API-KEY: ${TYPESENSE_API_KEY}" -d '{
          "id": "124",
          "company_name": ["Stark Industries", 123],
          "num_employees": 5215,
          "country": "USA"
        }'
```

## Expected Behavior

```
{"message": "Field `company_name` must be a string array."}
```

## Actual Behavior

```
terminate called after throwing an instance of 'nlohmann::detail::type_error'
  what():  [json.exception.type_error.302] type must be string, but is number
E20210217 19:35:12.100222     1 backward.hpp:4196] Stack trace (most recent call last):
E20210217 19:35:12.100255     1 backward.hpp:4196] #21   Object "/opt/typesense-server", at 0x86ab30, in _start
E20210217 19:35:12.100256     1 backward.hpp:4196] #20   Object "/lib/x86_64-linux-gnu/libc-2.23.so", at 0x7f4a7f66d83f, in __libc_start_main
E20210217 19:35:12.100258     1 backward.hpp:4196] #19   Source "/typesense/src/main/typesense_server.cpp", line 124, in main [0x7b1d1f]
E20210217 19:35:12.100260     1 backward.hpp:4196] #18   Source "/typesense/src/typesense_server_utils.cpp", line 408, in run_server [0x98ae92]
E20210217 19:35:12.100260     1 backward.hpp:4196] #17   Source "/typesense/src/http_server.cpp", line 195, in run [0x92f720]
E20210217 19:35:12.100262     1 backward.hpp:4196] #16   Source "/typesense/external-Linux/h2o-5accc794404996fb056f1826f5a172693de95f22/lib/common/socket/evloop.c.h", line 619, in h2o_evloop_run [0x9aa0d4]
E20210217 19:35:12.100263     1 backward.hpp:4196] #15   Source "/typesense/external-Linux/h2o-5accc794404996fb056f1826f5a172693de95f22/lib/common/socket/evloop.c.h", line 567, in run_pending [0x9a92e5]
E20210217 19:35:12.100265     1 backward.hpp:4196] #14 | Source "/typesense/external-Linux/h2o-5accc794404996fb056f1826f5a172693de95f22/lib/common/socket/evloop.c.h", line 534, in read_on_ready
E20210217 19:35:12.100266     1 backward.hpp:4196]       Source "/typesense/external-Linux/h2o-5accc794404996fb056f1826f5a172693de95f22/lib/common/socket/evloop.c.h", line 248, in run_socket [0x9a91f8]
E20210217 19:35:12.100267     1 backward.hpp:4196] #13 | Source "/typesense/external-Linux/h2o-5accc794404996fb056f1826f5a172693de95f22/lib/common/multithread.c", line 90, in queue_cb
E20210217 19:35:12.100270     1 backward.hpp:4196]       Source "/typesense/external-Linux/h2o-5accc794404996fb056f1826f5a172693de95f22/lib/common/multithread.c", line 64, in h2o_buffer_consume [0x9a782d]
E20210217 19:35:12.100270     1 backward.hpp:4196] #12   Source "/typesense/include/http_data.h", line 412, in on_message [0x9347eb]
E20210217 19:35:12.100271     1 backward.hpp:4196] #11   Source "/typesense/src/core_api.cpp", line 1115, in raft_write_send_response [0x916899]
E20210217 19:35:12.100273     1 backward.hpp:4196] #10   Source "/typesense/src/core_api.cpp", line 596, in post_add_document [0x919a1c]
E20210217 19:35:12.100275     1 backward.hpp:4196] #9    Source "/typesense/src/collection.cpp", line 201, in add [0x8c5dfe]
E20210217 19:35:12.100275     1 backward.hpp:4196] #8    Source "/typesense/src/collection.cpp", line 293, in add_many [0x8c5941]
E20210217 19:35:12.100276     1 backward.hpp:4196] #7    Source "/typesense/src/collection.cpp", line 315, in batch_index [0x8c3f1b]
E20210217 19:35:12.100278     1 backward.hpp:4196] #6  | Source "/typesense/src/collection.cpp", line 406, in get
E20210217 19:35:12.100279     1 backward.hpp:4196]     | Source "/usr/local/gcc-10.1.0/include/c++/10.1.0/future", line 799, in _M_get_result
E20210217 19:35:12.100281     1 backward.hpp:4196]       Source "/usr/local/gcc-10.1.0/include/c++/10.1.0/future", line 722, in par_index_in_memory [0x8a9c21]
E20210217 19:35:12.100282     1 backward.hpp:4196] #5    Source "../../../../libstdc++-v3/libsupc++/eh_ptr.cc", line 249, in rethrow_exception [0x12ecc1a]
E20210217 19:35:12.100283     1 backward.hpp:4196] #4    Source "../../../../libstdc++-v3/libsupc++/eh_terminate.cc", line 58, in terminate [0x12ecc90]
E20210217 19:35:12.100284     1 backward.hpp:4196] #3    Source "../../../../libstdc++-v3/libsupc++/eh_terminate.cc", line 48, in abort [0x12ecc25]
E20210217 19:35:12.100286     1 backward.hpp:4196] #2    Source "../../../../libstdc++-v3/libsupc++/vterminate.cc", line 95, in __verbose_terminate_handler [0x7a06da]
E20210217 19:35:12.100287     1 backward.hpp:4196] #1    Object "/lib/x86_64-linux-gnu/libc-2.23.so", at 0x7f4a7f684039, in abort
E20210217 19:35:12.100288     1 backward.hpp:4196] #0    Object "/lib/x86_64-linux-gnu/libc-2.23.so", at 0x7f4a7f682438, in raise
Aborted (Signal sent by tkill() 1 0)
```

## Metadata

**Typsense Version**: typesense/typesense:0.19.0 from Docker

**OS**: Ubuntu 20.04.1 LTS (x86_64) with 5.4.0-65-generic kernel

**Comment by kishorenc** [comment]
Thanks for reporting this issue. Currently, Typesense only checks the first value of an array for type validation. We should change this behavior to check all values, even though it will have a bit of a performance impact on records having large number of array values.

**Comment by kishorenc** [comment]
@jabagawee I've a fix for this in this build: `typesense/typesense:0.20.0.rc11` -- can you please test as well to confirm?

**Comment by jabagawee** [comment]
perfect!

**Comment by jabagawee** [comment]
(i've confirmed that the fix works)

**Comment by kishorenc** [comment]
Addressed as part of `v0.20.0` released last week.
